lucet-runtime/src/c_api.rs

```diff
@@ -507,6 +508,15 @@ pub unsafe fn lucet_vmctx_yield(vmctx: &Vmctx, val: *mut c_void) -> *mut c_void
         .unwrap_or(std::ptr::null_mut())
 }
 
+/// TODO: do we need the `lucet_hostcall` macro invocation here, which performs interesting
+/// contortions related to panic-unwind catching, if this is a frequent invocation and we are
+/// performing only a stack swap / control transfer to the host and back?
```

Comment:
yes, but more because `lucet_hostcall!` also takes care of signalling the appropriate timeout mechanism. so without a `lucet_hostcall!` a guest timeout while yielded would SIGALRM the embedder!

Reply:
This should be temporary, though, since the bounded execution mechanism should be able to replace the uses of `SIGALRM`. That shouldn't happen until a followup PR, so it's right to use `#[lucet_hostcall]` here